### PR TITLE
[ci] switch from MiKTeX to tinytex on Windows R jobs (fixes #5600)

### LINKED ISSUE #5600: [ci] [R-package] Miktex installation failing: "Timeout was reached"
## Description

R Windows CI jobs on multiple PRs and `master` with the following error.

> Setting up MiKTeX
miktexsetup_standalone: Timeout was reached
miktexsetup_standalone: Data: code="28", url="https://api2.miktex.org/hello"
Error: Process completed with exit code -1.

## Reproducible example

Example recent build from #5396: https://github.com/microsoft/LightGBM/actions/runs/3528457186/jobs/5918579958

## Environment info

Four CI jobs are failing regularly with this error:

* `r-package (windows-latest, MINGW, R 3.6, cran)`
* `r-package (windows-latest, MINGW, R 4.2, cran)`
* `r-package (windows-latest, MINGW, R 3.6, cmake)`
* `r-package (windows-latest, MINGW, R 4.2, cmake)`

## Additional Comments

In the past, we've seen MikTeX issues like this with minimal error messages resolve themselves after 12-48 hours 🙃 

* https://github.com/microsoft/LightGBM/issues/5562
* https://github.com/microsoft/LightGBM/issues/4005
* https://github.com/microsoft/LightGBM/issues/3454
* https://github.com/microsoft/LightGBM/issues/3198

### PR DESCRIPTION
**Pull request by jameslamb**
Fixes #5600.

R CI jobs require a LaTeX distribution to build the package documentation. This PR proposes switching from [MiKTeX](https://miktex.org/) to [`{tinytex}`](https://yihui.org/tinytex/) for Windows builds.

## Motivation

The use of MiKTeX has been the source of several multi-day outages in LightGBM's CI over the last few years (and most recently, #5600):

* https://github.com/microsoft/LightGBM/issues/5562
* https://github.com/microsoft/LightGBM/issues/4005
* https://github.com/microsoft/LightGBM/issues/3454
* https://github.com/microsoft/LightGBM/issues/3198

Several of those have had the resolution "wait a few days until changes in CTAN mirrors or other upstream sources are fixed". That's very disruptive to our development here, as it means either skipping R package CI jobs or not merging anything for multiple days.

I'm hoping that because `{tinytex}` is lighter-weight, it'll be more stable.

I also believe using the GitHub Action https://github.com/r-lib/actions/tree/v2-branch/setup-tinytex will remove the maintenance burden of keeping up with MiKTeX and other updates upstream, since that action is actively maintained.

For more, see ["Why TinyTex?"](https://yihui.org/tinytex/pain/) in the `{tinytex}` docs.

### CODE REVIEW

**Comment by jameslamb** [comment]
@guolinke the next time you have a few minutes, can you please review this PR? This will unblock CI, which will unblock #5396, which will prevent CI from breaking starting 8 days from now when Azure removes the `macos-10.15` environment.

I'm sorry to keep `@`-ing you, but there is a lot that needs to get done here, there hasn't been much activity recently from other maintainers, and I am trying to get the project into a good state before some planned travel starting next week.

**Comment by guolinke** [comment]
@jameslamb no problems, please feel free to @ me when you need any helps.

**Review by guolinke — APPROVED** [review]

**Comment by jameslamb** [comment]
Thank you so much! I'll merge this, merge `master` into #5396, and then merge #5396 once it builds successfully.

**Pull request merged**
